### PR TITLE
POL-105-Azure Blob Storage Optimization

### PR DESCRIPTION
**Pull request by chandravurubindi**
POL-105-Azure Blob Storage Optimization

### Description

Checks Azure Blob Storage for last updated time and moves the object to the Cool or Archive tier after user approval

### Issues Resolved

- This policy identifies all Azure Blob Storage objects last modified outside of the specified time frame
- For all objects identified as old, the user can choose to move the object to Cool or Archived tiers 
  after user approval

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### Running Link: 
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5d82190d84fd710089f3b46f

### CODE REVIEW

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+parameter "param_azure_storage_account" do
+  type "string"
+  label "Name of Storage account"
+  category "Azure"
+  description "Name of Storage account to search for blobs"
+end
+
+parameter "param_cooltier_days" do
+  type "string"
+  label "Days since last modified to move to cooltier"
```
Comment (cdwilhelm) [inline cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: should be Cool tier

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+  description "Move to cooltier after days last modified - leave blank to skip moving"
+end
+
+parameter "param_archivetier_days" do
+  type "string"
+  label "Days since last modified to move to archivetier"
```
Comment (cdwilhelm) [inline cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: should be Archive tier

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+  type "string"
+  label "Name of Storage account"
+  category "Azure"
+  description "Name of Storage account to search for blobs"
+end
+
+parameter "param_cooltier_days" do
+  type "string"
+  label "Days since last modified to move to cooltier"
+  description "Move to cooltier after days last modified - leave blank to skip moving"
```
Comment (cdwilhelm) [inline cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: should be Cool tier

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+end
+
+parameter "param_archivetier_days" do
+  type "string"
+  label "Days since last modified to move to archivetier"
+  description "Move to archivetier after days last modified- leave blank to skip moving"
```
Comment (cdwilhelm) [inline cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: should be Archive tier

File: cost/azure/object_storage_optimization/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+### Prerequesites
+
+- Azure Service Principal (AKA Azure Active Directory Application) with the appropriate permissions to manage resources in the target subscription
+- The following RightScale Credentials
+  - `AZURE_APPLICATION_ID`
+  - `AZURE_APPLICATION_KEY`
+  
+### Required RightScale Roles
+
```
Comment (cdwilhelm) [inline cost/azure/object_storage_optimization/README.md]: add credential_viewer, and observer.

Comment (chandravurubindi) [reply cost/azure/object_storage_optimization/README.md]: Updated readme.md file with the permissions.

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt (hunk 4 above)
Comment (chandravurubindi) [reply cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: Thanks, I updated it now.

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt (hunk 3 above)
Comment (chandravurubindi) [reply cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: Thanks, I updated it now.

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt (hunk 2 above)
Comment (chandravurubindi) [reply cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: Updated it..

File: cost/azure/object_storage_optimization/azure_object_storage_optimization.pt (hunk 1 above)
Comment (chandravurubindi) [reply cost/azure/object_storage_optimization/azure_object_storage_optimization.pt]: Thanks, I updated it now.

**Comment by cdwilhelm** [comment]
@chandravurubindi  this should get fixed so we can merge friday. it's critical it's done this week.

**Pull request merged**